.github/workflows/main.yml

```diff
@@ -155,4 +159,4 @@ jobs:
       - uses: codecov/codecov-action@0565863a31f2c772f9f0395002a31e3f06189574 # v5.4.0
         with:
           fail_ci_if_error: false
-          files: coverage-no-integration/.coverage,coverage-integration/.coverage
+          files: coverage-no-integration-ubuntu-latest/.coverage,coverage-integration-ubuntu-latest/.coverage,coverage-no-integration-macos-latest/.coverage,coverage-integration-macos-latest/.coverage
```

Comment:
It is unfortunate that we cannot use the parameters in the matrix here. But I don't think it's a big deal if we need to fix two places. Could you please add a comment that when fixing the matrix, one should fix the names here as well?

Reply:
Makes sense to me. I think this could be simplified by running `coverage combine` over the various coverage files. Let me see if that works, as the current approach is indeed prone to mishaps, like you suggest.